### ISSUE TITLE
Visual Shader compile error with CanvasShaderGLES2

### ISSUE THREAD

**Issue by vixelz** [godotengine/godot#25839]
**Godot version:**
3.1beta4

**OS/device including version:**
Windows 10, GTX670

**Issue description:**
CanvasItem visual shaders throw compilation errors even for trivial shaders.

**Steps to reproduce:**

I created a new ShaderMaterial for a TextureRect, and I created a trivial visual shader, passing a zero vector to vertex in the vertex shader, and a zero vector as color in the fragment shader. Instead of this shader compiling I get the following shader compilation error:

    drivers/gles2/shader_gles2.cpp:196 - CanvasShaderGLES2: Fragment shader compilation failed:
    0(102) : error C0000: syntax error, unexpected '{' at token "{"
    0(130) : error C0000: syntax error, unexpected reserved word "if" at token "if"
    
    drivers/gles2/shader_gles2.cpp:431 - Method/Function Failed, returning: __null
    drivers/gles2/shader_gles2.cpp:133 - Condition ' !version ' is true. returned: false

Full output: [compile_output.txt](https://github.com/godotengine/godot/files/2858181/compile_output.txt)

**Minimal reproduction project:**

[VisualShaderDemo.zip](https://github.com/godotengine/godot/files/2858180/VisualShaderDemo.zip)

**Comment by vixelz** [comment]
Whoops, initially marked it as 3.1alpha4, I meant 3.1beta4

**Comment by clayjohn** [comment]
I can confirm this bug on linux with integrated intel graphics. Although, the error I get is slightly different

```
drivers/gles2/shader_gles2.cpp:196 - CanvasShaderGLES2: Fragment shader compilation failed:
0:102(1): error: syntax error, unexpected '{', expecting $end
```
And line 102 (for a bit of context):
```
99: uniform bool use_default_normal;
100: 
101: /* clang-format off */
102: {
103: }
104: 
105: 
106: /* clang-format on */
107: 
108: void main() {
```

The issue persists no matter what is in the shader.

**Comment by akien-mga** [comment]
I confirm that it's reproducible on 3.1 beta 4, and seems fixed already in 55ca2a7. Probably by one of @JFonS' recent PRs.
